### PR TITLE
Working 4.2.0 ebuilds tested on pytorch 1.9.0

### PR DESCRIPTION
**Pull request by fuag15**
Ok, this is tested against libtorch / pytorch 1.9.0 and working on a vega 64 (vega10 / gfx900). MIOpen keeps printing Info messages but aside from that things are good.

### CODE REVIEW

**Comment by fuag15** [comment]
i fixed the miopen issues and i think this is good to go as much as it can be without other testers

**Comment by justxi** [comment]
Except the too comments all looks good. 
I would accept it, if the llvm-roc ebuild is renamed.

File: sci-libs/miopen/miopen-4.0.0-r1.ebuild [hunk 1]
```diff
@@ -25,8 +25,8 @@ RDEPEND="
 	!opencl? (
 		=dev-util/rocm-clang-ocl-${PV}*
 		sci-libs/rocBLAS[tensile] )
-	!static-boost? ( =dev-libs/boost-1.72* )
-	static-boost? ( =dev-libs/boost-1.72*[static-libs] )"
+	!static-boost? ( >=dev-libs/boost-1.72 )
```
Comment (littlewu2508) [inline sci-libs/miopen/miopen-4.0.0-r1.ebuild:28]: [MIOpen requires boost-1.72](https://github.com/ROCmSoftwarePlatform/MIOpen/#prerequisites) to perform kernel cache, and I tried other versions which result in build failure. Can you show me your build.log for MIOpen?

**Review by littlewu2508 — COMMENTED** [review]
https://github.com/gentoo/gentoo/blob/master/dev-util/roctracer/roctracer-4.0.0-r1.ebuild is th official ebuild in gentoo portage, which is more concise. Better to be based on that version

**Comment by fuag15** [comment]
@justxi  roger that I'll clean things up, I'll have time next week. I've been talking with the rocm folks and it seems like it would be less of a headache ot put everything in /usrr/rocm or something like that and let rocm live in its own environ.

File: sci-libs/miopen/miopen-4.0.0-r1.ebuild (hunk 1 above)
Comment (fuag15) [reply sci-libs/miopen/miopen-4.0.0-r1.ebuild:28]: @littlewu2508 boost devs have a workaround built into boost for the issues it causes, see this patch:

https://github.com/fuag15/rocm/blob/master/sci-libs/miopen/files/miopen-4.2.0-disable-no-inline-boost.patch

defining that removes the offending preprocessor directives that were confusing llvm-roc

**Comment by justxi** [comment]
@fuag15 We add that discussion already where to install ROCm, but at the end we decided to keep the installation according to the Gentoo and FHS rules.

**Comment by justxi** [comment]
Please reopen if you have changed the suggested points. 
Thanks.

**Pull request closed without merging**